qiskit/primitives/containers/sampler_pub.py

```diff
@@ -0,0 +1,100 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""
+Sampler Pub class
+"""
+
+from __future__ import annotations
+
+from typing import Tuple, Union
+
+from qiskit import QuantumCircuit
+
+from .bindings_array import BindingsArray, BindingsArrayLike
+from .shape import ShapedMixin
+
+
+class SamplerPub(ShapedMixin):
+    """Pub (Primitive Unified Bloc) for Sampler.
+
+    Pub is composed of double (circuit, parameter_values).
+    """
+
+    __slots__ = ("_circuit", "_parameter_values")
+
+    def __init__(
+        self,
+        circuit: QuantumCircuit,
+        parameter_values: BindingsArray | None = None,
+        validate: bool = False,
+    ):
+        """Initialize a sampler pub.
+
+        Args:
+            circuit: a quantum circuit.
+            parameter_values: a bindings array.
+            validate: if True, the input data is validated during initialization.
+        """
```

Comment:
According to lint, the following seems necessary.
```suggestion
        """
        super().__init__()
```